cl/scrapers/management/commands/cl_scrape_ia.py

```diff
@@ -0,0 +1,196 @@
+import re
+import argparse
+import requests
+from internetarchive import get_files
+
+from cl.lib.command_utils import VerboseCommand, logger
+from cl.search.models import Opinion, OpinionCluster, Docket, Citation
+
+def find_docket_no_section(str):
+    """
+    This function attempts to find the docket No section that is often
+    not on the first page.  This is then used to parse out the docket nos.
+
+    :param str:
+    :return:
+    """
+    regex = r"Docket No.*.Filed|Docket No.*.(, [0-9]{4}.)"
+    matches = re.finditer(regex, str)
+    r = r"[0-9]{3,5}-[0-9A-Za-z]{2,4}(\.)"
+    for matchNum, match in enumerate(matches, start=1):
+        xst = str[match.start():]
+        matches2 = re.finditer(r, str[match.start():])
+        for matchNum, match in enumerate(matches2, start=1):
+            return xst[:match.end()]
+
+def get_docket_numbers(parsed_text):
+    """
+    Once the section has been parsed this regex grabs all of the docket nos.
+
+    :param parsed_text:
+    :return:
+    """
+    regex = r"[0-9]{3,5}-[0-9A-Za-z]{2,4}(\,|\.)"
+    matches = re.finditer(regex, parsed_text, re.MULTILINE)
+    hits = []
+    for matchNum, match in enumerate(matches, start=1):
+        hits.append(match.group())
+    docket_string = ", ".join(hits).replace(",,", ",")
+    return docket_string
+
+def get_citation(str):
+    """
+    This function grabs the first citation looking string and then cleans out
+    whitespace issues.  Then returns None (roughly 20/8500 times) or
+    a dictionary of the citation.
+
+    :param str:
+    :return:
+    """
+    regex = r".{10}(T\.).*|.{10}(T\. C\.).{10}|(T\.).*|(T\. C\.).{10}"
+    matches = re.finditer(regex, str)
+
+    for matchNum, match in enumerate(matches, start=1):
+        cite = match.group()
+        cd = {}
+        cite = cite.replace("T. C.", "T.C.").replace("Memo ", "Memo. ").replace("   ", " ").strip()
+        if "T.C. Memo." in cite:
+            cd['volume'] = cite.split("Memo.")[1].split("-")[0]
+            cd['reporter'] = "T.C. Memo."
+            cd['page'] = cite.split("Memo.")[1].split("-")[1]
+            cd['type'] = 8
+            return cd
+        else:
+            try:
+                c = cite.split(" T.C. No. ")
+                cd['volume'] = c[0].strip()
+                cd['reporter'] = "T.C."
+                cd['page'] = c[1].strip()
+                cd['type'] = 4
+
+                return cd
+            except:
+                return None
+    return None
+
+def update_tax_opinions(options):
+    """
+    This function can be run as a cronjob to add docket and case information to
+    scraped tax court cases.
+
+    :param options:
+    :return:
+    """
+
+    tax_dockets = Docket.objects.all().filter(court_id="tax").\
```

Comment:
Seems like this could be refactored as `OpinionCluster.objects.filter(docket__court='tax')`?